vmtest-haskell

```diff
@@ -1,12 +1,22 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
-KRUN_BINARY=".build/k/k-distribution/target/release/k/bin/krun"
-KOMPILED_DEF=".build/haskell/"
-HASKELL_BACKEND_CMD=".build/kore/bin/kore-exec"
-
-cSCHEDULE=DEFAULT
-cMODE=VMTESTS
+set -euo pipefail
 
 PGM=$1
 
-$KRUN_BINARY --directory "$KOMPILED_DEF" -cSCHEDULE="$cSCHEDULE" -cMODE="$cMODE" "$PGM" --haskell-backend-command "$HASKELL_BACKEND_CMD"
+assembled="$(mktemp)"
+kast="$(mktemp)"
+output="$(mktemp)"
+trap "rm -rf $kast $output $assembled" INT TERM EXIT
+"$(dirname "$0")/assemble-iele-test" "$PGM" > "$assembled"
+"$(dirname "$0")/kore-json.py" "$assembled" '`DEFAULT`(.KList)' '`VMTESTS`(.KList)' > "$kast"
+
+exit=0
+kore-exec .build/haskell/iele-testing-kompiled/definition.kore --pattern "$kast" --module IELE-TESTING --smt none --output "$output" || exit=$?
+
+[[ $exit == 0 ]] && exit 0
```

Comment:
This does not work with `set -e`, because if `$exit != 0`, the first command fails, then the whole line fails, then the whole program exits. So you need to use `||` when you have `set -e`, and invert the conditions.
```suggestion
[[ $exit != 0 ]] || exit 0
```